src/Umbraco.Web.UI.Client/src/views/propertyeditors/grid/editors/media.controller.js

```diff
@@ -1,112 +1,129 @@
                     $scope.model.config.startNodeId = userData.startMediaIds.length !== 1 ? -1 : userData.startMediaIds[0];
                     $scope.model.config.startNodeIsVirtual = userData.startMediaIds.length !== 1;
                 });
             }
         }
-        
-        $scope.setImage = function(){
+
+        $scope.setImage = function() {
             var startNodeId = $scope.model.config && $scope.model.config.startNodeId ? $scope.model.config.startNodeId : undefined;
-            var startNodeIsVirtual = startNodeId ? $scope.model.config.startNodeIsVirtual : undefined;
-            var value = $scope.control.value;
-            var target = value
-                ? {
-                    udi: value.udi,
-                    url: value.image,
-                    image: value.image,
-                    focalPoint: value.focalPoint,
-                    coordinates: value.coordinates
-                }
-                : null;
+
             var mediaPicker = {
                 startNodeId: startNodeId,
-                startNodeIsVirtual: startNodeIsVirtual,
+                startNodeIsVirtual: startNodeId ? $scope.model.config.startNodeIsVirtual : undefined,
                 cropSize: $scope.control.editor.config && $scope.control.editor.config.size ? $scope.control.editor.config.size : undefined,
                 showDetails: true,
                 disableFolderSelect: true,
                 onlyImages: true,
                 dataTypeKey: $scope.model.dataTypeKey,
-                currentTarget: target,
-                submit: function(model) {
-                    var selectedImage = model.selection[0];
-                   
-                    $scope.control.value = {
-                        focalPoint: selectedImage.focalPoint,
-                        coordinates: selectedImage.coordinates,
-                        id: selectedImage.id,
-                        udi: selectedImage.udi,
-                        image: selectedImage.image,
-                        caption: selectedImage.altText
-                    };
-                    
+                submit: model => {
+                    updateControlValue(model.selection[0]);                    
                     editorService.close();
                 },
-                close: function() {
-                    editorService.close();
-                }
-            }
-            
+                close: () => editorService.close()                
+            };
+
             editorService.mediaPicker(mediaPicker);
         };
+
+        $scope.editImage = function() {               
+
+            const mediaCropDetailsConfig = {
+                size: 'small',
+                target: $scope.control.value,
+                submit: model => {
+                    updateControlValue(model.target);
+                    editorService.close();
+                },
+                close: function () {
```

Comment:
@nathanwoulfe any reason you don't use the new format `()` for close function, but for submit function? 😁

Reply:
no good reason... I'll blame the lack of javascript linting...

Reply:
Tidied these up too